cmd/courier/watch.go

```diff
@@ -0,0 +1,73 @@
+package courier
+
+import (
+	cx "context"
+	"net/http"
+
+	"github.com/gorilla/context"
+	"github.com/spf13/cobra"
+	"github.com/urfave/negroni"
+
+	"github.com/ory/graceful"
+	"github.com/ory/kratos/driver"
+	"github.com/ory/kratos/x"
+	"github.com/ory/x/configx"
+	"github.com/ory/x/reqlog"
+)
+
+var watchCmd = &cobra.Command{
+	Use:   "watch",
+	Short: "Starts the ORY Kratos message courier",
+	Run: func(cmd *cobra.Command, args []string) {
+		d := driver.New(cmd.Context(), configx.WithFlags(cmd.Flags()))
+
+		go ServeMetrics(d, cmd, args)
+		Watch(d, cmd, args)
+	},
+}
+
+func ServeMetrics(r driver.Registry, cmd *cobra.Command, args []string) {
+
+	c := r.Config(cmd.Context())
+	l := r.Logger()
+	n := negroni.New()
+
+	router := x.NewRouterAdmin()
```

Comment:
We can keep the metrics around if you have a concrete use case for it (e.g. watching the metrics for this task). However, I would like to make it optional. One issue is also that this is using the admin config port, which might cause issues when not running as containers but on bare-metal (port re-allocation). It would need several config files, and so on - so how about adding a CLI flag `--expose-metrics-port 123` which, if set, enables the metrics port?

Reply:
Perfect, I’ll add this.